maestro-cli/src/main/java/maestro/cli/command/UploadCommand.kt

```diff
@@ -0,0 +1,221 @@
+        } else {
+            message("Existing auth token is invalid or expired")
+            cachedAuthTokenFile.deleteIfExists()
+            null
+        }
+    }
+
+    private fun setCachedAuthToken(token: String?) {
+        cachedAuthTokenFile.parent.createDirectories()
+        if (token == null) {
+            cachedAuthTokenFile.deleteIfExists()
+        } else {
+            cachedAuthTokenFile.writeText(token)
+        }
+    }
+
+    private fun triggerSignInFlow(): String {
+        message("No auth token found")
+        val email = prompt("Sign In or Sign Up using your company email address:")
+        var isLogin = true
+        val requestToken = client.magicLinkLogin(email, AUTH_SUCCESS_REDIRECT_URL).getOrElse { loginError ->
+            if (loginError.code == 403 && loginError.body?.string()?.contains("not an authorized email address") == true) {
+                isLogin = false
+                message("No existing team found for this email domain")
+                val team = prompt("Enter a team name to create your team:")
+                client.magicLinkSignUp(email, team, AUTH_SUCCESS_REDIRECT_URL).getOrElse { signUpError ->
+                    throw CliError(signUpError.body?.string() ?: signUpError.message)
+                }
+            } else {
+                throw CliError(loginError.body?.string() ?: loginError.message)
+            }
+        }
+
+        if (isLogin) {
+            message("We sent a login link to $email. Click on the link there to finish logging in...")
+        } else {
+            message("We sent an email to $email. Click on the link there to finish creating your account...")
+        }
+
+        while (true) {
+            val errResponse = when (val result = client.magicLinkGetToken(requestToken)) {
+                is Ok -> {
+                    if (isLogin) {
+                        message("✅ Login successful")
+                    } else {
+                        message("✅ Team created successfully")
+                    }
+                    setCachedAuthToken(result.value)
+                    return result.value
+                }
+                is Err -> result.error
+            }
+            val errorMessage = errResponse.body?.string() ?: errResponse.message
+            if (
+                "Login process not complete" !in errorMessage
+                && "Email is not authorized" !in errorMessage
+            ) {
+                throw CliError("Failed to get auth token (${errResponse.code}): $errorMessage")
+            }
+            Thread.sleep(1000)
+        }
+    }
+
+    companion object {
+
+        private const val AUTH_SUCCESS_REDIRECT_URL = "https://console.mobile.dev/auth/success"
+
+        private fun message(message: String) {
+            println(ansi().render("@|cyan \n$message |@"))
+        }
+
+        private fun prompt(message: String): String {
+            print(ansi().render("\n@|yellow,bold $message\n> |@"))
```

Comment:
minor nit: `ansi` has nice DSL for setting colours

Reply:
This is actually using the jansi convenience syntax already. The actual ansi codes are much uglier 😄